openfisca_core/entities/_core_entity.py

```diff
@@ -72,11 +73,46 @@ def get_variable(
             None: When the ``Variable`` doesn't exist.
 
         Raises:
+            ValueError: When the :attr:`_tax_benefit_system` is not set yet.
             ValueError: When ``check_existence`` is ``True`` and
                 the ``Variable`` doesn't exist.
 
-        """
+        Examples:
+            >>> from openfisca_core import (
+            ...     entities,
+            ...     periods,
+            ...     taxbenefitsystems,
+            ...     variables,
+            ... )
+
+            >>> this = entities.SingleEntity("this", "", "", "")
+            >>> that = entities.SingleEntity("that", "", "", "")
+
+            >>> this.get_variable("tax")
+            Traceback (most recent call last):
+            ValueError: You must set 'tax_benefit_system' before calling thi...
+
+            >>> tax_benefit_system = taxbenefitsystems.TaxBenefitSystem([this])
+            >>> this.set_tax_benefit_system(tax_benefit_system)
 
+            >>> this.get_variable("tax")
+
+            >>> this.get_variable("tax", check_existence=True)
+            Traceback (most recent call last):
+            VariableNotFoundError: You tried to calculate or to set a value...
+
+            >>> class tax(variables.Variable):
+            ...     definition_period = periods.MONTH
+            ...     value_type = float
+            ...     entity = that
```

Comment:
Why using different entities, `this` and `that`  here ?

Reply:
To show that this method is going to get you the variable regardless of who is this variable being defined for. It was a confusing behaviour for me (a couple of years ago I implemented an actual filter, to search within the variables defined for the entity, but I'll repropose that one later as it is a breaking change).

Reply:
(The test in itself has exactly that purpose, for people to think "Wait, this is weird behaviour". If tomorrow someone changes this, the test will fail, and that is what I actually expect).